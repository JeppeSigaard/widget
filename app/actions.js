import { BASEURL, APPID } from "../config";
import axios from "axios";
import { stringify } from "query-string";

// Action constant
export const SET_WEATHER = "SET_WEATHER";

// Set weather
export const setWeather = function(status, data) {
  return { type: SET_WEATHER, status, data };
};

// Get weather
export const getWeather = function(city) {
  return (dispatch, getState) => {
    const query = stringify({ q: `${city},dk`, appid: APPID, units: "metric" });
    dispatch(setWeather("loading"));

    axios
      .get(`${BASEURL}?${query}`)
      .then(response => {
        dispatch(setWeather("success", response.data));
      })
      .catch(e => {
        dispatch(setWeather("failed", e));
      });
  };
};
