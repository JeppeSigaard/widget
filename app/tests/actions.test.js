import * as actions from "../actions";

describe("Action creator", () => {
  it("has action constant", () => {
    expect(actions.SET_WEATHER).toMatchSnapshot();
  });

  it("creates an action to set weather", () => {
    expect(actions.setWeather("loading")).toMatchSnapshot();
    expect(actions.setWeather("failed")).toMatchSnapshot();
    expect(actions.setWeather("success", { sum: "data" })).toMatchSnapshot();
  });
});
