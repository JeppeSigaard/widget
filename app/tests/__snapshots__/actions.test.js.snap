// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action creator creates an action to set weather 1`] = `
Object {
  "data": undefined,
  "status": "loading",
  "type": "SET_WEATHER",
}
`;

exports[`Action creator creates an action to set weather 2`] = `
Object {
  "data": undefined,
  "status": "failed",
  "type": "SET_WEATHER",
}
`;

exports[`Action creator creates an action to set weather 3`] = `
Object {
  "data": Object {
    "sum": "data",
  },
  "status": "success",
  "type": "SET_WEATHER",
}
`;

exports[`Action creator has action constant 1`] = `"SET_WEATHER"`;
